src/NuGet.Core/NuGet.RuntimeModel/JsonRuntimeFormat.cs

```diff
@@ -62,81 +61,104 @@ public static RuntimeGraph ReadRuntimeGraph(JToken json)
         }
 
-        private static void WriteRuntimeDescription(JObject jObject, RuntimeDescription data)
+        private static void WriteRuntimeDescription(IObjectWriter writer, RuntimeDescription data)
         {
-            var value = new JObject();
-            jObject[data.RuntimeIdentifier] = value;
-            value["#import"] = new JArray(data.InheritedRuntimes.Select(x => new JValue(x)));
-            foreach (var x in data.RuntimeDependencySets.Values)
+            writer.WriteObjectStart(data.RuntimeIdentifier);
+
+            writer.WriteNameArray("#import", data.InheritedRuntimes);
+
+            var sortedDependencySets = data.RuntimeDependencySets
+                                           .OrderBy(pair => pair.Key, StringComparer.Ordinal)
+                                           .Select(pair => pair.Value);
+
+            foreach (var set in sortedDependencySets)
             {
-                WriteRuntimeDependencySet(value, x);
+                WriteRuntimeDependencySet(writer, set);
             }
+
+            writer.WriteObjectEnd();
         }
 
-        private static void WriteRuntimeDependencySet(JObject jObject, RuntimeDependencySet data)
+        private static void WriteRuntimeDependencySet(IObjectWriter writer, RuntimeDependencySet data)
         {
-            var value = new JObject();
-            jObject[data.Id] = value;
-            foreach (var x in data.Dependencies.Values)
+            writer.WriteObjectStart(data.Id);
+
+            var sortedDependencies = data.Dependencies
+                                         .OrderBy(pair => pair.Key, StringComparer.Ordinal)
+                                         .Select(pair => pair.Value);
+
+            foreach (var dependency in sortedDependencies)
             {
-                WritePackageDependency(value, x);
+                WritePackageDependency(writer, dependency);
             }
+
+            writer.WriteObjectEnd();
         }
 
-        private static void WritePackageDependency(JObject jObject, RuntimePackageDependency data)
+        private static void WritePackageDependency(IObjectWriter writer, RuntimePackageDependency data)
         {
-            jObject[data.Id] = new JValue(data.VersionRange.ToNormalizedString());
+            writer.WriteNameValue(data.Id, data.VersionRange.ToNormalizedString());
         }
 
-        private static void WriteCompatibilityProfile(JObject jObject, CompatibilityProfile data)
+        private static void WriteCompatibilityProfile(IObjectWriter writer, CompatibilityProfile data)
         {
-            var value = new JObject();
-            jObject[data.Name] = value;
-            foreach(var frameworkGroup in data.RestoreContexts.GroupBy(f => f.Framework))
+            writer.WriteObjectStart(data.Name);
+
+            var sortedFrameworkGroups = data.RestoreContexts.GroupBy(context => context.Framework)
+                                            .OrderBy(pair => pair.Key.DotNetFrameworkName);
```

Comment:
Why this order by on some property? We have a couple comparers for `NuGetFramework` that should be used. Try `NuGetFrameworkSorter`.